Zend/tests/closure_const_expr/attributes_scope_002.phpt

```diff
@@ -0,0 +1,35 @@
+--TEST--
+Closure in attribute may not access unrelated private variables
+--EXTENSIONS--
+reflection
+--FILE--
+<?php
+
+#[Attribute(Attribute::TARGET_CLASS | Attribute::IS_REPEATABLE)]
+class Attr {
+    public function __construct(public Closure $value) {}
+}
+
+#[Attr(static function (E $e) {
+    echo $e->secret, PHP_EOL;
+})]
+class C {
+}
```

Comment:
this looks super scary. do you have a use-case for this construction?

Reply:
What construction are you referring to?

Reply:
using a closure from within a attribute

Reply:
Yes. Custom validation for an attribute-based validation library:

```
class Foo
{
  #[Validate\Custom(function (string $value) {
    return $value !== '';
  })]
  public string $bar;
}
```

Reply:
I see. having used this example in the test, would make feel it less syntetic :).

Reply:
This PR is a Draft and not cleaned up yet. It will get a proper description and commit history. And most likely this also requires an RFC. For now I was just interested in whether this is going to work in theory.